### PR TITLE
✨ Create a new pom file for freeze

### PR DESCRIPTION
**Pull request by yogyagamage**
Fixes
- #688 
- #697 

The freeze functionality is updated such that,
    - When freeze is executed a new pom file will be created with the default name `pom.lockfile.xml`. A different name can be used with `-DpomLockfileName` 
    - The versions for the transitive dependencies will be pinned by defining them inside `dependencyManagement` instead of declaring them as direct dependencies. Scopes will also be pinned.

### CODE REVIEW

File: maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/FreezeDependencyMojo.java [hunk 1]
```diff
@@ -41,41 +42,75 @@ public class FreezeDependencyMojo extends AbstractMojo {
     @Parameter(defaultValue = "${session}", readonly = true, required = true)
     private MavenSession session;
 
+    @Parameter(defaultValue = "pom.lockfile.xml", property = "pomLockfileName")
+    private String pomLockfileName;
```
Comment (algomaster99) [inline maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/FreezeDependencyMojo.java:46]:
```suggestion
    @Parameter(defaultValue = "pom.lockfile.xml", property = "pomLockfileOutput")
    private String pomLockfileOutput;
```

Just following convention as most CLIs have the output flag to create an output file.

File: maven_plugin/src/test/java/it/IntegrationTestsIT.java [hunk 2]
```diff
@@ -99,11 +99,20 @@ public void freezeJunit(MavenExecutionResult result) throws Exception {
     }
 
     @MavenTest
-    public void checkSpringFreeze(MavenExecutionResult result) throws Exception {
+    public void freezeWithoutDepManagement(MavenExecutionResult result) throws Exception {
```
Comment (algomaster99) [inline maven_plugin/src/test/java/it/IntegrationTestsIT.java:102]: Can you also add an assertion to verify that the original POM is not modified?

**Review by algomaster99 — CHANGES_REQUESTED** [review]
Looks good! Suggested minor changes.

**Comment by monperrus** [comment]
:100:

**Pull request merged**